src/debug_tools/common.rs

```diff
@@ -0,0 +1,120 @@
+use std::fmt::{self, Display};
+
+use crate::{
+    nibbles::{Nibble, Nibbles},
+    partial_trie::{Node, PartialTrie},
+    utils::TrieNodeType,
+};
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub(super) enum PathSegment {
+    Empty,
+    Hash,
+    Branch(Nibble),
+    Extension(Nibbles),
+    Leaf(Nibbles),
+}
+
+impl Display for PathSegment {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            PathSegment::Empty => write!(f, "Empty"),
+            PathSegment::Hash => write!(f, "Hash"),
+            PathSegment::Branch(nib) => write!(f, "Branch({})", nib),
+            PathSegment::Extension(nibs) => write!(f, "Extension({})", nibs),
+            PathSegment::Leaf(nibs) => write!(f, "Leaf({})", nibs),
+        }
+    }
+}
+
+impl PathSegment {
+    pub(super) fn node_type(&self) -> TrieNodeType {
+        match self {
+            PathSegment::Empty => TrieNodeType::Empty,
+            PathSegment::Hash => TrieNodeType::Hash,
+            PathSegment::Branch(_) => TrieNodeType::Branch,
+            PathSegment::Extension(_) => TrieNodeType::Extension,
+            PathSegment::Leaf(_) => TrieNodeType::Leaf,
+        }
+    }
+
+    pub(super) fn get_key_piece_from_seg_if_present(&self) -> Option<Nibbles> {
+        match self {
+            PathSegment::Empty | PathSegment::Hash => None,
+            PathSegment::Branch(nib) => Some(Nibbles::from_nibble(*nib)),
+            PathSegment::Extension(nibs) => Some(*nibs),
+            PathSegment::Leaf(nibs) => Some(*nibs),
```

Comment:
nit (same as above):
```suggestion
            PathSegment::Extension(nibs) | PathSegment::Leaf(nibs) => Some(*nibs),
```